main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -976,38 +974,38 @@ object GenExpression {
       visitor.visitFieldInsn(PUTSTATIC, declaration, field.getName, JvmOps.getJvmType(exp.tpe).toDescriptor)
 
       // Push Unit on the stack.
-      visitor.visitFieldInsn(GETSTATIC, JvmName.Unit.toInternalName, GenUnitClass.InstanceFieldName, JvmName.Unit.toDescriptor)
+      visitor.visitFieldInsn(GETSTATIC, BackendObjType.Unit.jvmName.toInternalName, GenUnitClass.InstanceFieldName, BackendObjType.Unit.jvmName.toDescriptor)
 
     case Expression.NewChannel(exp, _, loc) =>
       addSourceLine(visitor, loc)
-      visitor.visitTypeInsn(NEW, JvmName.Channel.toInternalName)
+      visitor.visitTypeInsn(NEW, BackendObjType.Channel(null).jvmName.toInternalName)
       visitor.visitInsn(DUP)
       compileExpression(exp, visitor, currentClass, lenv0, entryPoint)
-      visitor.visitMethodInsn(INVOKESPECIAL, JvmName.Channel.toInternalName, "<init>", AsmOps.getMethodDescriptor(List(JvmType.PrimInt), JvmType.Void), false)
+      visitor.visitMethodInsn(INVOKESPECIAL, BackendObjType.Channel(null).jvmName.toInternalName, "<init>", AsmOps.getMethodDescriptor(List(JvmType.PrimInt), JvmType.Void), false)
```

Comment:
We never use null. That leads to null pointer exceptions!

Reply:
ill add JvmName.Channel back and phase it out as I go through everything